### PR TITLE
rebuild usage of realtime information

### PR DESCRIPTION
**Pull request by ue71603**
We now have three levels:
* none - only timetable data
* explanatory - realtime is used, but cancelled and disrupted journeys are still shown. 
* full - realtime information is used inf full, cancellation, blocking, consequences are taken into account

### CODE REVIEW

**Review by Aurige — DISMISSED** [review]

File: OJP/OJP_JourneySupport.xsd [hunk 1]
```diff
@@ -516,9 +535,9 @@
 				</xs:element>
 				<xs:group ref="NumberOfResultsGroup" minOccurs="0"/>
 			</xs:choice>
-			<xs:element name="IgnoreRealtimeData" type="xs:boolean" default="false" minOccurs="0">
+			<xs:element name="UseRealtimeData" type="UseRealtimeDataEnumeration" default="full" minOccurs="0">
 				<xs:annotation>
-					<xs:documentation>The trip calculation should not use any realtime or incident data.</xs:documentation>
+					<xs:documentation>The way realtime data is used in the calculation of the trip (repleases IgnoreRealtimeData)</xs:documentation>
```
Comment (sgrossberndt) [inline OJP/OJP_JourneySupport.xsd:540]: `<xs:documentation>The way realtime data should be used in the calculation of the trip</xs:documentation>`

Regarding " (repleases IgnoreRealtimeData)": We did not add "historical" info about renamed/removed elements to the xsd yet, this rather belongs to the changes document. Even if this should be added, it should be "replaces" instead.

File: OJP/OJP_JourneySupport.xsd [hunk 2]
```diff
@@ -503,6 +503,25 @@
...
+	<xs:simpleType name="UseRealtimeDataEnumeration">
+		<xs:restriction base="xs:string">
+			<xs:enumeration value="full">
+				<xs:annotation>
+					<xs:documentation>Full usage of realtime information, including the removing of SERVICE JOURNEYS.</xs:documentation>
+				</xs:annotation>
+			</xs:enumeration>
+			<xs:enumeration value="explanatory">
+				<xs:annotation>
+					<xs:documentation>Cancelled SERVICE JOURNEY are still being shown.</xs:documentation>
```
Comment (sgrossberndt) [inline OJP/OJP_JourneySupport.xsd:515]: `<xs:documentation>Cancelled SERVICE JOURNEYs are still displayed</xs:documentation>`

Or maybe "delivered" or "returned" instead?

File: OJP/OJP_JourneySupport.xsd [hunk 3]
```diff
@@ -503,6 +503,25 @@
 			</xs:element>
 		</xs:sequence>
 	</xs:group>
+	<xs:simpleType name="UseRealtimeDataEnumeration">
+		<xs:restriction base="xs:string">
+			<xs:enumeration value="full">
+				<xs:annotation>
+					<xs:documentation>Full usage of realtime information, including the removing of SERVICE JOURNEYS.</xs:documentation>
```
Comment (sgrossberndt) [inline OJP/OJP_JourneySupport.xsd:510]: `<xs:documentation>Full use of real-time information, including removal of SERVICE JOURNEYS.</xs:documentation>`

**Review by sgrossberndt — DISMISSED** [review]

**Comment by herlitze** [comment]
> * . When a vehicle is late and connections is not working, then the trip is not shown

Why would you not show these?

File: OJP/OJP_JourneySupport.xsd (hunk 3 above)
Comment (ue71603) [reply OJP/OJP_JourneySupport.xsd:510]: changed

File: OJP/OJP_JourneySupport.xsd (hunk 2 above)
Comment (ue71603) [reply OJP/OJP_JourneySupport.xsd:515]: changed.

File: OJP/OJP_JourneySupport.xsd (hunk 1 above)
Comment (ue71603) [reply OJP/OJP_JourneySupport.xsd:540]: changed

File: OJP/OJP_Trips.xsd [hunk 4]
```diff
@@ -1009,6 +1009,11 @@
 					<xs:documentation>Refers to a leg to be refined by the server.</xs:documentation>
 				</xs:annotation>
 			</xs:element>
+			<xs:element name="UseRealtimeData" type="UseRealtimeDataEnumeration" default="full" minOccurs="0">
+				<xs:annotation>
+					<xs:documentation>Usage of realtime data in refinment. Default is "fulL".</xs:documentation>
```
[the remark this answers is not in the document]
Comment (ue71603) [reply OJP/OJP_Trips.xsd:1014]: changed

**Comment by ue71603** [comment]
> > * . When a vehicle is late and connections is not working, then the trip is not shown
> 
> Why would you not show these?

changed it in the description. Was no problem in xsd

File: OJP/OJP_Trips.xsd [hunk 5]
```diff
@@ -1009,6 +1009,11 @@
 					<xs:documentation>Refers to a leg to be refined by the server.</xs:documentation>
 				</xs:annotation>
 			</xs:element>
+			<xs:element name="UseRealtimeData" type="UseRealtimeDataEnumeration" default="full" minOccurs="0">
+				<xs:annotation>
+					<xs:documentation>Usage of realtime data in refinement. Default is "fulL".</xs:documentation>
```
Comment (skinkie) [inline OJP/OJP_Trips.xsd:1014]:
```suggestion
					<xs:documentation>Usage of realtime data in refinement. Default is "full".</xs:documentation>
```

File: OJP/OJP_TripInfo.xsd [hunk 6]
```diff
@@ -38,9 +38,9 @@
 			<xs:documentation>Parameters that control the response behaviour.</xs:documentation>
 		</xs:annotation>
 		<xs:sequence>
-			<xs:element name="UseTimetabledDataOnly" type="xs:boolean" default="false" minOccurs="0">
+			<xs:element name="UseRealTimeData" type="UseRealtimeDataEnumeration" default="full" minOccurs="0">
 				<xs:annotation>
-					<xs:documentation>Do not show any realtime or incident data. Default is false.</xs:documentation>
+					<xs:documentation>Use realtime data. Default is "fulL"</xs:documentation>
```
Comment (skinkie) [inline OJP/OJP_TripInfo.xsd:43]:
```suggestion
					<xs:documentation>Use realtime data. Default is "full"</xs:documentation>
```

File: OJP/OJP_StopEvents.xsd [hunk 7]
```diff
@@ -119,9 +119,9 @@
 					<xs:documentation>Whether operating day information of this stop event should be included in the response.</xs:documentation>
 				</xs:annotation>
 			</xs:element>
-			<xs:element name="IncludeRealtimeData" type="xs:boolean" default="false" minOccurs="0">
+			<xs:element name="UseRealtimeData" type="UseRealtimeDataEnumeration" default="none" minOccurs="0">
 				<xs:annotation>
-					<xs:documentation>Whether realtime information of this stop event should be included in the response.</xs:documentation>
+					<xs:documentation>Whether realtime information of this stop event should be used in the response. Default is "none"</xs:documentation>
```
Comment (skinkie) [inline OJP/OJP_StopEvents.xsd:124]: Why would it be "none" here? Departure display mode sounds like a proper default for me.

**Review by ue71603 — COMMENTED** [review]
changed

File: OJP/OJP_StopEvents.xsd (hunk 7 above)
Comment (ue71603) [reply OJP/OJP_StopEvents.xsd]: changed

**Review by herlitze — DISMISSED** [review]

**Review by Aurige — DISMISSED** [review]

**Review by trurlurl — CHANGES_REQUESTED** [review]
StopEventResult supports error code STOPEVENT_NOREALTIME, TripResult and MultiTrip support TRIP_REALTIMEINCOMPLETE. Shouldn't similar codes be added for TripInfoResult and RefineResult?

**Comment by ue71603** [comment]
will be adapted

**Comment by ue71603** [comment]
> StopEventResult supports error code STOPEVENT_NOREALTIME, TripResult and MultiTrip support TRIP_REALTIMEINCOMPLETE. Shouldn't similar codes be added for TripInfoResult and RefineResult?

Done for TripInfo. The rest we need to do in the end.

**Review by trurlurl — DISMISSED** [review]

File: OJP/OJP_StopEvents.xsd [hunk 8]
```diff
@@ -119,9 +119,9 @@
 					<xs:documentation>Whether operating day information of this stop event should be included in the response.</xs:documentation>
 				</xs:annotation>
 			</xs:element>
-			<xs:element name="IncludeRealtimeData" type="xs:boolean" default="false" minOccurs="0">
+			<xs:element name="UseRealtimeData" type="UseRealtimeDataEnumeration" default="full" minOccurs="0">
 				<xs:annotation>
-					<xs:documentation>Whether realtime information of this stop event should be included in the response.</xs:documentation>
+					<xs:documentation>Whether realtime information of this stop event should be used in the response. Default is "full"</xs:documentation>
```
Comment (skinkie) [inline OJP/OJP_StopEvents.xsd:124]:
```suggestion
					<xs:documentation>Whether real-time information of this stop event should be used in the response. Default is "full"</xs:documentation>
```

File: OJP/OJP_TripInfo.xsd [hunk 9]
```diff
@@ -38,9 +38,9 @@
 			<xs:documentation>Parameters that control the response behaviour.</xs:documentation>
 		</xs:annotation>
 		<xs:sequence>
-			<xs:element name="UseTimetabledDataOnly" type="xs:boolean" default="false" minOccurs="0">
+			<xs:element name="UseRealTimeData" type="UseRealtimeDataEnumeration" default="full" minOccurs="0">
 				<xs:annotation>
-					<xs:documentation>Do not show any realtime or incident data. Default is false.</xs:documentation>
+					<xs:documentation>Use realtime data. Default is "full"</xs:documentation>
```
Comment (skinkie) [inline OJP/OJP_TripInfo.xsd:43]:
```suggestion
					<xs:documentation>Use real-time data. Default is "full"</xs:documentation>
```

File: OJP/OJP_TripInfo.xsd [hunk 10]
```diff
@@ -175,6 +175,11 @@
 					<xs:documentation>No geographic information available for this vehicle journey.</xs:documentation>
 				</xs:annotation>
 			</xs:enumeration>
+			<xs:enumeration value="TRIPINFO_NOREALTIME">
+				<xs:annotation>
+					<xs:documentation>No real time information available.</xs:documentation>
```
Comment (skinkie) [inline OJP/OJP_TripInfo.xsd:180]:
```suggestion
					<xs:documentation>No real-time information available.</xs:documentation>
```

File: OJP/OJP_Trips.xsd [hunk 11]
```diff
@@ -1077,6 +1077,11 @@
 					<xs:documentation>Refers to a leg to be refined by the server.</xs:documentation>
 				</xs:annotation>
 			</xs:element>
+			<xs:element name="UseRealtimeData" type="UseRealtimeDataEnumeration" default="full" minOccurs="0">
+				<xs:annotation>
+					<xs:documentation>Usage of realtime data in refinement. Default is "full".</xs:documentation>
```
Comment (skinkie) [inline OJP/OJP_Trips.xsd:1082]:
```suggestion
					<xs:documentation>Usage of real-time data in refinement. Default is "full".</xs:documentation>
```

File: OJP/OJP_JourneySupport.xsd [hunk 12]
```diff
@@ -531,6 +531,25 @@
...
+	<xs:simpleType name="UseRealtimeDataEnumeration">
+		<xs:restriction base="xs:string">
+			<xs:enumeration value="full">
+				<xs:annotation>
+					<xs:documentation>Full use of real-time information, including removal of SERVICE JOURNEYS.</xs:documentation>
+				</xs:annotation>
+			</xs:enumeration>
+			<xs:enumeration value="explanatory">
+				<xs:annotation>
+					<xs:documentation>Cancelled and delayed SERVICE JOURNEYs are still returned, but an additional explanatory textual information is provide to decribe their current realtime status.</xs:documentation>
```
Comment (skinkie) [inline OJP/OJP_JourneySupport.xsd:543]:
```suggestion
					<xs:documentation>Cancelled and delayed SERVICE JOURNEYs are still returned, but an additional explanatory textual information is provide to decribe their current real-time status.</xs:documentation>
```

File: OJP/OJP_JourneySupport.xsd [hunk 13]
```diff
@@ -544,9 +563,9 @@
 				</xs:element>
 				<xs:group ref="NumberOfResultsGroup" minOccurs="0"/>
 			</xs:choice>
-			<xs:element name="IgnoreRealtimeData" type="xs:boolean" default="false" minOccurs="0">
+			<xs:element name="UseRealtimeData" type="UseRealtimeDataEnumeration" default="full" minOccurs="0">
 				<xs:annotation>
-					<xs:documentation>The trip calculation should not use any realtime or incident data.</xs:documentation>
+					<xs:documentation>The way realtime data should be used in the calculation of the trip.</xs:documentation>
```
Comment (skinkie) [inline OJP/OJP_JourneySupport.xsd:568]:
```suggestion
					<xs:documentation>The way real-time data should be used in the calculation of the trip.</xs:documentation>
```

**Review by skinkie — DISMISSED** [review]

**Review by skinkie — DISMISSED** [review]

**Review by Aurige — DISMISSED** [review]

**Review by trurlurl — APPROVED** [review]

**Review by trurlurl — DISMISSED** [review]

**Review by skinkie — APPROVED** [review]

**Review by AndreasAtSBB — APPROVED** [review]
Change: Real-time ... to Default is "full" Inluding HTML doc change.

**Pull request merged**